src/js/components/Animate.js

```diff
@@ -0,0 +1,203 @@
+// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP
+
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import TransitionGroup from 'react-addons-transition-group';
+import classnames from 'classnames';
+
+const CLASS_ROOT = 'animate';
+
+class AnimateChild extends Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      enterClass: '',
+      leaveClass: ''
+    };
+  }
+
+  componentWillAppear (callback) {
+    this.enter(callback);
+  }
+
+  componentWillEnter (callback) {
+    this.enter(callback);
+  }
+
+  componentDidAppear () {
+    this.entered();
+  }
+
+  componentDidEnter () {
+    this.entered();
+  }
+
+  componentWillLeave (callback) {
+    this.leave(callback);
+  }
+
+  enter (callback) {
+    const {
+      enter: {
+        animation: enterAnimation,
+        delay
+      },
+      leave: {
+        animation: leaveAnimation
+      }
+    } = this.props;
+    const node = ReactDOM.findDOMNode(this);
+
+    const enterClass = enterAnimation;
+    const leaveClass = leaveAnimation || enterAnimation;
+    this.setState({enterClass});
+    this.setState({leaveClass});
+
+    if (node) {
+      node.style.transitionDuration = '';
+      node.classList.remove('animate', `${leaveClass}--leave`);
+      node.classList.add(`${enterClass}--enter`);
+      setTimeout(callback, delay);
+    }
+  }
+
+  entered () {
+    const { enter: { duration } } = this.props;
+    const node = ReactDOM.findDOMNode(this);
+    const enterClass = this.state.enterClass;
+    node.classList.remove(`${enterClass}--enter`);
+    if (duration) {
+      node.style.transitionDuration = `${duration}ms`;
+    }
+    node.classList.add('animate', `${enterClass}--enter-active`);
+    setTimeout(() => {
+      node.style.transitionDuration = '';
+      node.classList.remove('animate');
+    }, parseFloat(getComputedStyle(node).transitionDuration) * 1000);
+  }
+
+  leave (callback) {
+    const { leave: { duration, delay } } = this.props;
+    const node = ReactDOM.findDOMNode(this);
+
+    if (duration) {
+      node.style.transitionDuration = `${duration}ms`;
+    }
+
+    return setTimeout(() => {
+      node.classList.remove(`${this.state.enterClass}--enter-active`);
+      node.classList.add('animate', `${this.state.leaveClass}--leave`);
+      setTimeout(callback, parseFloat(getComputedStyle(node).transitionDuration) * 1000);
+    }, delay);
+  }
+
+  render () {
+    return this.props.children;
+  }
+};
+
```

Comment:
Can you add AnimateChild.propTypes so we can see what the properties interface is?